pkg/operator/staticpod/controller/missingstaticpod/missing_static_pod_controller.go

```diff
@@ -0,0 +1,263 @@
+	configMapListerForTargetNamespace corelisterv1.ConfigMapNamespaceLister
+	targetNamespace                   string
+	operandName                       string
+}
+
+// New checks the latest installer pods.  If the installer pod was successful, and if it has
+// been longer than the terminationGracePeriodSeconds+10seconds since the installer pod
+// completed successfully, and if the static pod is not at the correct revision, this
+// controller will go degraded.
+// It will also emit an event for detection in CI,
+func New(
+	operatorClient v1helpers.StaticPodOperatorClient,
+	kubeInformersForTargetNamespace informers.SharedInformerFactory,
+	eventRecorder events.Recorder,
+	targetNamespace string,
+	staticPodName string,
+) factory.Controller {
+	c := &missingStaticPodController{
+		operatorClient:                    operatorClient,
+		podLister:                         kubeInformersForTargetNamespace.Core().V1().Pods().Lister(),
+		configMapListerForTargetNamespace: kubeInformersForTargetNamespace.Core().V1().ConfigMaps().Lister().ConfigMaps(targetNamespace),
+		targetNamespace:                   targetNamespace,
+		operandName:                       staticPodName,
+	}
+	return factory.New().
+		ResyncEvery(time.Minute).
+		WithInformers(
+			operatorClient.Informer(),
+			kubeInformersForTargetNamespace.Core().V1().Pods().Informer(),
+		).
+		WithSync(c.sync).
+		WithSyncDegradedOnError(operatorClient).
+		ToController("MissingStaticPodController", eventRecorder)
+}
+
+func (c *missingStaticPodController) sync(ctx context.Context, syncCtx factory.SyncContext) error {
+	_, originalOperatorStatus, _, err := c.operatorClient.GetStaticPodOperatorState()
+	if err != nil {
+		return err
+	}
+
+	installerPods, err := c.podLister.List(labels.SelectorFromSet(labels.Set{"app": "installer"}))
+	if err != nil {
+		return err
+	}
+
+	// get the most recent installer pod for each node
+	latestInstallerPodsByNode, err := getMostRecentInstallerPodByNode(installerPods)
+	if err != nil {
+		return err
+	}
+
+	errors := make([]string, 0, len(latestInstallerPodsByNode))
+	for node, latestInstallerPodOnNode := range latestInstallerPodsByNode {
+		installerPodRevision, err := getInstallerPodRevision(latestInstallerPodOnNode)
+		if err != nil {
+			// we expect every installer pod to have a installerPodRevision in its name, unexpected error here
+			return fmt.Errorf("failed to get installerPodRevision for installer pod %q - %w", latestInstallerPodOnNode.Name, err)
+		}
+
+		finishedAt, ok := installerPodFinishedAt(latestInstallerPodOnNode)
+		if !ok {
+			// either it's in the process of running, or it ran into an error
+			continue
+		}
+
+		staticPodRevisionOnThisNode := getStaticPodCurrentRevision(node, originalOperatorStatus)
+		gracePeriod, err := c.getStaticPodTerminationGracePeriodSecondsForRevision(staticPodRevisionOnThisNode)
+		if err != nil {
+			return err
+		}
+		threshold := time.Duration(gracePeriod) + 10*time.Second
```

Comment:
Should we consider increasing the margin? 10 seconds seems a bit low, increasing it to 1 minute might be safer and we shouldn't miss any error since once a static pod is missing, we can expect it not to appear for a while.

Reply:
let's wait until the proof pr for `kas-o` finishes